src/workflows-app/SubmissionHistory.test.js

```diff
@@ -0,0 +1,561 @@
+    // * ascending order
+    await user.click(await within(headers[headerPosition.Status]).getByRole('button'));
+    within(topRowCells(headerPosition.Status)).getByText('Success');
+
+    // * descending order
+    await user.click(await within(headers[headerPosition.Status]).getByRole('button'));
+    within(topRowCells(headerPosition.Status)).getByText('Failed with 1 errors');
+
+    // Click on "Duration" column and check that the top column is correct for:
+    // * ascending order
+    await user.click(await within(headers[headerPosition.Duration]).getByRole('button'));
+    within(topRowCells(headerPosition.Duration)).getByText('1 day 1 hour 1 minute 1 second');
+
+    // * descending order
+    await user.click(await within(headers[headerPosition.Duration]).getByRole('button'));
+    within(topRowCells(headerPosition.Duration)).getByText('1 month 1 day 1 hour 1 minute 1 second');
+  });
+
+  it('should display no content message when there are no previous run sets', async () => {
+    // Arrange
+    const mockRunSetResponse = jest.fn(() => Promise.resolve([]));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: mockRunSetResponse,
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    // Assert
+    await waitFor(() => {
+      expect(mockRunSetResponse).toBeCalled();
+    });
+
+    const table = screen.getByRole('table');
+    expect(table).toHaveAttribute('aria-colcount', '6');
+    expect(table).toHaveAttribute('aria-rowcount', '1');
+
+    const rows = within(table).getAllByRole('cell');
+    within(rows[1]).findByText('Nothing here yet! Your previously run submissions will be displayed here.');
+  });
+
+  it('should correctly display previous 2 run sets', async () => {
+    const getRunSetsMethod = jest.fn(() => Promise.resolve(runSetData));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: jest.fn(getRunSetsMethod),
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+
+    const table = screen.getByRole('table');
```

Comment:
`waitFor` and `getBy...` could be replaced by `findBy...`

```suggestion
    const table = await screen.findByRole('table');
```

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find